stubs/Reflection.phpstub

```diff
@@ -76,6 +76,17 @@ class ReflectionProperty implements Reflector
      * @return array<ReflectionAttribute<TClass>>
      */
     public function getAttributes(?string $name = null, int $flags = 0): array {}
+
+    /**
+     * @since 7.4
+     * @psalm-assert-if-true ReflectionType $this->getType()
+     */
+    public function hasType() : bool {}
+
+    /**
+     * @since 7.4
+     */
+    public function getType() : ?ReflectionType {}
```

Comment:
I'm not sure what test fails, but maybe you need to be defensive here and only mention the new (?!) type in the PHPDoc:

```suggestion
     * @return ReflectionType|null
     */
    public function getType() {}
```

Reply:
Never mind, just saw now that `ReflectionParameter` uses the same:

https://github.com/vimeo/psalm/blob/f381fba070c9a005359c21209ab85517a544e1e9/stubs/Reflection.phpstub#L112